### PR TITLE
Consider `FileResponse.chunk_size` when handling multiple ranges

### PR DESCRIPTION
**Pull request by Kludex**
cc @frostming

### CODE REVIEW

File: starlette/responses.py [hunk 1]
```diff
@@ -419,10 +413,14 @@ async def _handle_multiple_ranges(
...
-                    await file.seek(start)
-                    chunk = await file.read(min(self.chunk_size, end - start))
                     await send({"type": "http.response.body", "body": header_generator(start, end), "more_body": True})
-                    await send({"type": "http.response.body", "body": chunk, "more_body": True})
+                    while True:
+                        await file.seek(start)
+                        chunk = await file.read(min(self.chunk_size, end - start))
+                        start += len(chunk)
+                        await send({"type": "http.response.body", "body": chunk, "more_body": True})
+                        if len(chunk) < self.chunk_size or start >= end:
```
Comment (Kludex) [inline starlette/responses.py]: Is the `len(chunk) < self.chunk_size` necessary?

I'm too tired, I'll check tomorrow. Just leaving my own comment here.

**Comment by Kludex** [comment]
How do I have a different behavior in Python versions with this diff? 🤔

**Comment by Kludex** [comment]
> How do I have a different behavior in Python versions with this diff? 🤔

Coverage is not able to understand that the `break` was reached... @graingert do you know why? 🤔

File: starlette/responses.py [hunk 2]
```diff
@@ -419,10 +413,14 @@ async def _handle_multiple_ranges(
...
+                            break
```
Comment (Kludex) [inline starlette/responses.py]:
```suggestion
                            break  # pragma: no cover
```

File: starlette/responses.py (hunk 1 above)
Comment (frostming) [reply starlette/responses.py]: it can be mathematically deduced that is not required

Comment (abersheeran) [reply starlette/responses.py]: Yes, there is no need to check once `len(chunk) < self.chunk_size`

File: starlette/responses.py [hunk 3]
```diff
@@ -419,10 +413,14 @@ async def _handle_multiple_ranges(
...
-                    chunk = await file.read(min(self.chunk_size, end - start))
                     await send({"type": "http.response.body", "body": header_generator(start, end), "more_body": True})
-                    await send({"type": "http.response.body", "body": chunk, "more_body": True})
+                    while True:
+                        await file.seek(start)
+                        chunk = await file.read(min(self.chunk_size, end - start))
+                        start += len(chunk)
+                        await send({"type": "http.response.body", "body": chunk, "more_body": True})
+                        if len(chunk) < self.chunk_size or start >= end:
+                            break  # pragma: no cover
```
Comment (Kludex) [inline starlette/responses.py]:
```suggestion
                    while start < end:
                        await file.seek(start)
                        chunk = await file.read(min(self.chunk_size, end - start))
                        start += len(chunk)
                        await send({"type": "http.response.body", "body": chunk, "more_body": True})
```

File: starlette/responses.py [hunk 4]
```diff
@@ -414,13 +414,11 @@ async def _handle_multiple_ranges(
             async with await anyio.open_file(self.path, mode="rb") as file:
                 for start, end in ranges:
                     await send({"type": "http.response.body", "body": header_generator(start, end), "more_body": True})
-                    while True:
+                    while start < end:
                         await file.seek(start)
```
Comment (frostming) [inline starlette/responses.py]:
```suggestion
                    await file.seek(start)
                    while start < end:
```
Don't need to seek in every loop

**Comment by Kludex** [comment]
@abersheeran @frostming anything else here, or we are good?

**Comment by frostming** [comment]
lgtm

**Pull request merged**